vmdb/spec/lib/miq_automation_engine/miq_ae_state_machine_new_spec.rb

```diff
@@ -0,0 +1,68 @@
+require 'spec_helper'
+require 'timecop'
+
+describe "MiqAeStateMachine" do
+  before do
+    DummyWorkspace = Class.new do
+      def initialize(options = {})
```

Comment:
@kbrock I don't really have a problem with the class setup, though I did talk to Madhu and say probably the only thing I'd do differently was to make the workspace a stub. I think I'd do it like this:

```ruby
class TestClass
  include MiqAeEngine::MiqAeStateMachine

  def initialize(workspace)
    @workspace = workspace
  end
end

describe "MiqAeStateMachine" do
  let(:workspace) { instance_double("MiqAeEngine::MiqAeWorkspace", :root => options) }
  let(:test_class) { TestClass.new(workspace) }

  describe "#enforce_max_retries" do
    context "when the options are blank" do
      let(:options) { {} }

      it "works" do
        expect { test_class.enforce_max_retries }.to_not raise_error
      end
    end

    context "when the max retries are exceeded" do
      let(:options) { {"ae_state_retries" => 3} }

      it "blows up" do
        expect { test_class.enforce_max_retries("max_retries" => 4) }.to raise_error
      end
    end

    # Other tests, etc.
  end
end
```